packages/storybook/src/stories/grid-sorting.stories.tsx

```diff
@@ -317,3 +339,110 @@ export const SortedEventsApi = () => {
     </React.Fragment>
   );
 };
+export const SortedEventsOptions = () => {
+  const rows = React.useMemo(() => getRows(), []);
+  const cols = React.useMemo(() => getColumns(), []);
+  const [loggedEvents, setEvents] = React.useState<any[]>([]);
+
+  const handleEvent = React.useCallback(
+    (name, params) => {
+      action(name)(params);
+      setEvents((prev: any[]) => [...prev, name]);
+    },
+    [setEvents],
+  );
+
+  const onColumnsSorted = React.useCallback((params) => handleEvent('ColumnsSorted', params), [
+    handleEvent,
+  ]);
+  const onColumnsSortingChange = React.useCallback(
+    (params) => handleEvent('ColumnsSortingChange', params),
+    [handleEvent],
+  );
+  const sortModel = React.useMemo(
+    () =>
+      [
+        { field: 'age', sort: 'desc' },
+        { field: 'name', sort: 'asc' },
+      ] as SortModel,
+    [],
+  );
+
+  // We had the ol so we can test it with image snapshots
+  return (
+    <React.Fragment>
+      <div>
+        <h1 style={{ fontSize: '16pt' }}>Triggered Events in order </h1>
+        <ol>{...loggedEvents.map((evt, idx) => <li key={evt + idx}>{evt}</li>)}</ol>
+      </div>
+      <div className="grid-container">
+        <XGrid
+          rows={rows}
+          columns={cols}
+          options={{
+            onColumnsSorted,
+            onColumnsSortingChange,
+            sortModel,
+          }}
+        />
+      </div>
+    </React.Fragment>
+  );
+};
+
+function sortServerRows(rows: any[], params: ColumnSortedParams): Promise<any[]> {
+  return new Promise<any[]>((resolve) => {
+    setTimeout(() => {
+      if (params.sortModel.length === 0) {
+        resolve(getRows());
+      }
+      const sortedCol = params.sortModel[0];
+      const comparator = params.sortedColumns[0].sortComparator!;
+      let sortedRows = [
+        ...rows.sort((a, b) => comparator(a[sortedCol.field], b[sortedCol.field], a, b)),
+      ];
+
+      if (params.sortModel[0].sort === 'desc') {
+        sortedRows = sortedRows.reverse();
+      }
+
+      resolve(sortedRows);
+    }, 500);
+  });
+}
+
+export const ServerSideSorting = () => {
+  const [rows, setRows] = React.useState<RowsProp>(getRows());
+  const [columns] = React.useState<ColDef[]>(getColumns());
+  const [isLoading, setLoading] = React.useState<boolean>(false);
+
+  const onColumnsSortingChange = React.useCallback(
+    async (params: ColumnSortedParams) => {
+      setLoading(true);
+      action('onColumnsSortingChange')(params);
+
+      const newRows = await sortServerRows(rows, params);
+      setRows(newRows);
+      setLoading(false);
+    },
+    [setLoading, rows, setRows],
+  );
+
+  const sortBy: SortModel = React.useMemo(() => [{ field: 'age', sort: 'desc' }], []);
```

Comment:
yes otherwise it's going to try to reapply the sort. We use immutability principles

Reply:
What about we add a comment somewhere? It might be something people need to be aware of. As a general rule, I think that anytime memoziation is required user-land, it's a great opportunity to mention it in the documentation or event in the prop description.